transport_docker.rb

```diff
@@ -0,0 +1,39 @@
+require 'bolt/transport/base'
+require 'bolt/transport/docker/connection'
+require 'open3'
+
+class BoltDocker < Bolt::Transport::Docker::Connection
+  attr_reader :target
+
+  # Adapted from Bolt::Transport::Docker::Connection.execute.
+  def execute(command, stdin: nil)
+    if target.options['shell-command'] && !target.options['shell-command'].empty?
+      # escape any double quotes in command
+      command = command.gsub('"', '\"')
+      command = "#{target.options['shell-command']} \" #{command}\""
+    end
+    command = *Shellwords.split(command)
+
+    command_options = []
+    # Need to be interactive if redirecting STDIN
+    command_options << '--interactive' unless stdin.nil?
+    command_options << '--tty' if target.options['tty']
+    command_options << container_id
+    command_options.concat(command)
+
+    env_hash = {}
+    # Set the DOCKER_HOST if we are using a non-default service-url
+    env_hash['DOCKER_HOST'] = @docker_host unless @docker_host.nil?
+
+    in_r, in_w = IO.pipe
+    in_w.sync = true
```

Comment:
What part of this streams to stdout/stderr?

Reply:
Spawn's default behavior is to stream to inherit file descriptors (stdin/stdout/stderr) from the parent.

Reply:
Ah ok. Thanks